transport/tcp/tcp.go

```diff
@@ -137,34 +147,49 @@ func (d *dialer) GetOption(n string) (interface{}, error) {
 }
 
 type listener struct {
-	addr     *net.TCPAddr
-	bound    net.Addr
-	proto    transport.ProtocolInfo
-	listener *net.TCPListener
-	opts     options
+	addr       *net.TCPAddr
+	bound      net.Addr
+	proto      transport.ProtocolInfo
+	listener   *net.TCPListener
+	opts       options
+	handshaker transport.Handshaker
 }
 
 func (l *listener) Accept() (transport.Pipe, error) {
 
 	if l.listener == nil {
 		return nil, mangos.ErrClosed
 	}
-	conn, err := l.listener.AcceptTCP()
-	if err != nil {
-		return nil, err
-	}
-	if err = l.opts.configTCP(conn); err != nil {
-		conn.Close()
-		return nil, err
-	}
-	return transport.NewConnPipe(conn, l.proto, l.opts)
+	return l.handshaker.Wait()
 }
 
 func (l *listener) Listen() (err error) {
 	l.listener, err = net.ListenTCP("tcp", l.addr)
-	if err == nil {
-		l.bound = l.listener.Addr()
+	if err != nil {
+		return
 	}
+	l.bound = l.listener.Addr()
+	go func() {
+		for {
+			conn, err := l.listener.AcceptTCP()
```

Comment:
When the listener is Close()d, the goroutine continues to run (leaks) - maybe exit goroutine on this condition (on Linux `EINVAL`, not sure about Windows).

Reply:
Same issue as you already identified.